db/dohsql.c

```diff
@@ -759,7 +759,14 @@ static int dohsql_dist_next_row(struct sqlclntstate *clnt, sqlite3_stmt *stmt)
         }
     }
     if (!empty) {
-        poll(NULL, 0, 10);
+        if (bdb_lock_desired(thedb->bdb_env)) {
```

Comment:
Maybe we could wrap the logic into a function so we don't repeat ourselves?

Reply:
It is just an if, I prefer that to a another wrapper on recover_deadlock